shared/lib/fetch-with-cache.ts

```diff
@@ -8,6 +8,7 @@ const fetchWithCache = async ({
   cacheOptions: { cacheRefreshTime = MINUTE * 6, timeout = SECOND * 30 } = {},
   functionName = '',
   allowStale = false,
+  cacheKey = `cachedFetch:${url}`,
```

Comment:
Since we pre-seed the "safeChainsList" cache (now known as "wellKnownChains") we needed to know the `cacheKey` on the outside in order to set it. Quite the leaky abstraction! Now we can pass in our own `cacheKey`, making this leaky abstraction a part of the API itself! Yay!

The abstraction is still leaky, as the `safeChainsList` pre-seed functionality requires knowing what APIs the `fetchWithCache` function uses internally, but at least it's now a little better, I guess?